include/boost/iterator/iterator_facade.hpp

```diff
@@ -811,11 +769,11 @@ namespace iterators {
         typename boost::iterators::detail::operator_brackets_result<Derived, Value, reference>::type
         operator[](difference_type n) const
         {
-            typedef boost::iterators::detail::use_operator_brackets_proxy<Value, Reference> use_proxy;
+            const auto use_proxy = boost::iterators::detail::use_operator_brackets_proxy<Value, Reference>::value;
 
             return boost::iterators::detail::make_operator_brackets_result<Derived>(
                 this->derived() + n
-              , use_proxy()
+              , std::integral_constant<bool, use_proxy>{}
             );
```

Comment:
I think, this should be better left as a `typedef` instead of introducing a local variable. Something like this:

```
using use_proxy = boost::iterators::detail::use_operator_brackets_proxy<Value, Reference>;

return boost::iterators::detail::make_operator_brackets_result<Derived>(
    ...
    , std::integral_constant<bool, use_proxy::value>{}
);
```

Technically, local variables, even `const` ones, are still allocated on the stack, and I would prefer `operator[]` to be as lightweight as possible.